### PR TITLE
[v6r15] Various fixes

### PR DESCRIPTION
**Pull request by fstagni**
- fixed double creation of db tables in SystemLoggingDB
- added some logging to RSS InfoGetter
- simplification of code for RSS InfoGetter
- made some tests discoverable by py.test
- added/reviving tests for ConsistencyInspector
- Jenkins/CI shell scripts: added some protections for navigations

### CODE REVIEW

File: ResourceStatusSystem/Utilities/InfoGetter.py [hunk 1]
```diff
@@ -1,217 +1,57 @@
...
-
-
-
 __RCSID__ = '$Id: $'
 
-class InfoGetter:
-  """
-    Class InfoGetter is in charge of getting information from the RSS Configurations
+
+def sanitizedecisionParams( decisionParams ):
```
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: why taking it out of the class ? It logically belongs there no ?

File: ResourceStatusSystem/Utilities/InfoGetter.py [hunk 2]
```diff
@@ -270,32 +115,29 @@ def __getPoliciesThatApply( self, decisionParams ):
 
       # We are not going to use name / type anymore, but we keep them for debugging
       # and future usage.
-      policyDict = {
-                     'name' : policyName,
+      policyDict = { 'name' : policyName,
                      'type' : policyType,
-                     'args' : {}
-                   }
+                     'args' : {} }
```
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: If we agree to follow pep8 as I think we did, the previous convention is the correct one

File: ResourceStatusSystem/Utilities/InfoGetter.py [hunk 3]
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
...
+      return S_ERROR("No domain names have been specified on the CS")
+    domainNames = domainNames[ 'Value' ]
 
-    # Get policies that match the given decissionParameters
-    for policyActionName, policyActionConfig in policyActionsConfig.items():
-      policyMatch = Utils.configMatch( decisionParams, policyActionConfig )
-      if policyMatch:
-        policyActionsThatApply.append( policyActionName )
+    unknownDomains = list( set(targetDomain) - set(domainNames) )
+    if len(unknownDomains) > 0:
```
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: ```if unknownDomains:```

File: ResourceStatusSystem/Utilities/InfoGetter.py [hunk 4]
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
...
+      gLogger.warn( "Domains %s belong to the policy parameters but not to the CS domains" % unknownDomains )
 
-    return S_OK( policyActionsThatApply )
+    knownDomains = list( set(domainNames) & set(targetDomain) )
+    if len(knownDomains) == 0:
```
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: ```if not knowDomains:```

File: ResourceStatusSystem/Utilities/InfoGetter.py [hunk 5]
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
...
+        if result['OK']:
+          ces = result['Value']
+          #to verify that the given CE is in the list of the LCG CEs
+          if name not in ces:
+            gLogger.info( "ComputingElement %s NOT found in domains %s" % ( name, domains )  )
+            return False
+          else:
+            gLogger.info( "ComputingElement %s found in domains %s" % ( name, domains ) )
+        else:
+          gLogger.warn( "unable to verify if ComputingElement %s is in domains %s" % ( name, domains ) )
```
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: no return false here ?

**Comment by chaen** [comment]
I don't see how commit 0fed8f6152bd5d6e7f892c8e55c987bfea2c5d64 fixes the double creation ? There is still nowhere a check before calling ``_createTables```

File: ResourceStatusSystem/Utilities/InfoGetter.py (hunk 1 above)
Comment (fstagni) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: To be honest I think the whole class is useless here. Just functions would be enough

**Comment by fstagni** [comment]
The tables creation was invoked both in the dB and in the service classes

**Comment by chaen** [comment]
I find terribly misleading that ```checkTable``` actually does ```createTable```. 
But are you sure that this is enough ? Next time you will restart the service, it will call ```checkTable``` again, so the problem is the same as calling it twice.

**Comment by fstagni** [comment]
Yes, this `checkTables` actually creates the tables when they are not there. In that same mentioned commit I removed the call to it from the service, so now it's only in the DB class. It's tested in certification. BTW I also updated with the suggested fixes.

**Comment by chaen** [comment]
Yes you are right, it works, but I don't understand why :-) I'll check the rest

File: ResourceStatusSystem/Utilities/InfoGetter.py (hunk 1 above)
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: Okay but then why doing it only for one method ? It seems more logical to do it for all or none

File: ResourceStatusSystem/Utilities/InfoGetter.py (hunk 2 above)
Comment (chaen) [inline ResourceStatusSystem/Utilities/InfoGetter.py]: Maybe went unnoticed, so I just poke that one

File: tests/Integration/Framework/Test_LoggingDB.py [hunk 6]
```diff
@@ -9,3 +9,131 @@
 
 
 DIRAC.exit()
```
Comment (chaen) [inline tests/Integration/Framework/Test_LoggingDB.py]: Unconditional exit, just like that ? :D

File: tests/Jenkins/dirac_ci.sh [hunk 7]
```diff
@@ -246,7 +246,10 @@ function DIRACPilotInstall(){
 	findRelease
 
 	#Don't launch the JobAgent here
+	cwd=$PWD
+	cd $PILOTINSTALLDIR
```
Comment (chaen) [inline tests/Jenkins/dirac_ci.sh]: for safety, maybe check that the cd was successful

File: tests/Jenkins/utilities.sh [hunk 8]
```diff
@@ -43,6 +43,25 @@ function default(){
...
+	currentBranch=`git --git-dir=$TESTCODE/DIRAC/.git rev-parse --abbrev-ref HEAD`
+
+	if [ $currentBranch == 'integration' ]
+	then
+		echo 'we were already on integration, no need to change'
+		# get the releases.cfg file
+		cp $TESTCODE/DIRAC/releases.cfg $TESTCODE/releases.cfg
+	else
+		cwd=$PWD
+		cd $TESTCODE/DIRAC/
```
Comment (chaen) [inline tests/Jenkins/utilities.sh]: test the rc

File: Resources/Storage/test/test_StorageElement.py [hunk 9]
```diff
@@ -4,8 +4,8 @@
```
Comment (atsareg) [inline Resources/Storage/test/test_StorageElement.py]: Should be renamed to Test_StorageElement.py ?

File: Resources/Storage/test/test_StorageFactory.py [hunk 10]
```diff
@@ -2,8 +2,8 @@
```
Comment (atsareg) [inline Resources/Storage/test/test_StorageFactory.py]: ibid

File: Resources/Storage/test/test_StoragePlugIn.py [hunk 11]
```diff
@@ -5,8 +5,8 @@
```
Comment (atsareg) [inline Resources/Storage/test/test_StoragePlugIn.py]: ibid

**Comment by atsareg** [comment]
Review OK

**Comment by fstagni** [comment]
I left the "test_" instead of "Test_" for some of the files because the tests in them just don't work as of now, and would just make Travis to fail. They need to be reviewed and changed, or moved.

**Comment by petricm** [comment]
Maybe we should distinguish somehow what is a pure python unit test and what needs additional infrastructure like CS, SE, etc. ?

**Comment by atsareg** [comment]
I thought that this is what should go to the Jenkins integration and tests in the systems

**Comment by fstagni** [comment]
Exactly: everything that needs something else than "pure python" goes to the tests/ directory. So, for example everything that needs a cfg, and that needs to start with `parseCommandLine()` goes there.

**Comment by fstagni** [comment]
For the tests in question, they are at the moment some abandonware that should be revisited, and maybe moved. For the moment, I left them with lower-case 't' because this way they won't be discovered and therefore won't be run by Travis.

**Pull request merged**